app/Http/Controllers/API/EventController.php

```diff
@@ -161,9 +161,18 @@ public function addVolunteer(Request $request, $idevents)
             $full_name = null;
         }
 
-        // User is null, this volunteer is either anonymous or no user exists.
+        $eventRole = Role::RESTARTER;
+
         if ($request->has('user') && $request->input('user') !== 'not-registered') {
+            // User is null, this volunteer is either anonymous or no user exists.
             $user = $request->input('user');
+
+            if ($user) {
+                $u = User::find($user);
+
+                // A host of the group who is added to an event becomes a host of the event.
+                $eventRole = $u && $u->role == Role::HOST && Fixometer::userIsHostOfGroup($party->group, $user) ? Role::HOST : Role::RESTARTER;
```

Comment:
I think this should just be

`$eventRole = $u && Fixometer::userIsHostOfGroup($party->group, $user) ? Role::HOST : Role::RESTARTER;`

as there could be admins or network coordinators who are host of the group, and should get set as host of event, even if their overall role might not be Role::HOST.